src/main/java/io/cdap/plugin/gcp/bigquery/common/BigQueryBaseConfig.java

```diff
@@ -30,113 +30,198 @@
-import io.cdap.plugin.gcp.common.GCPConfig;
 import io.cdap.plugin.gcp.common.GCPUtils;
 import io.cdap.plugin.gcp.gcs.GCSPath;
 
+import java.io.IOException;
 import javax.annotation.Nullable;
 
+
 /**
  * Common configuration class for BigQuery sources, sinks and engines.
  */
-public class BigQueryBaseConfig extends GCPConfig {
-  private static final String SCHEME = "gs://";
-
-  public static final String DATASET_PROJECT_ID = "datasetProject";
-  public static final String NAME_DATASET = "dataset";
-  public static final String NAME_BUCKET = "bucket";
-
-  @Name(DATASET_PROJECT_ID)
-  @Macro
-  @Nullable
-  @Description("The project in which the dataset is located/should be created."
-    + " Defaults to the project specified in the Project Id property.")
-  private String datasetProject;
-
-  @Name(NAME_DATASET)
-  @Macro
-  @Description("The dataset to write to. A dataset is contained within a specific project. "
-    + "Datasets are top-level containers that are used to organize and control access to tables and views.")
-  protected String dataset;
-
-  @Name(NAME_BUCKET)
-  @Macro
-  @Nullable
-  @Description("The Google Cloud Storage bucket to store temporary data in. "
-    + "Cloud Storage data will be deleted after it is loaded into BigQuery. "
-    + "If it is not provided, a unique bucket will be automatically created and then deleted after the run finishes. "
-    + "The service account must have permission to create buckets in the configured project.")
-  protected String bucket;
-
-  @Nullable
-  public String getDatasetProject() {
-    if (GCPConfig.AUTO_DETECT.equalsIgnoreCase(datasetProject)) {
-      return ServiceOptions.getDefaultProjectId();
+public class BigQueryBaseConfig extends PluginConfig {
+    private static final String SCHEME = "gs://";
+
+    public static final String NAME_DATASET = "dataset";
+    public static final String NAME_BUCKET = "bucket";
+    public static final String NAME_CMEK_KEY = "cmekKey";
+
+    @Name(NAME_DATASET)
+    @Macro
+    @Description("The dataset to write to. A dataset is contained within a specific project. "
+            + "Datasets are top-level containers that are used to organize and control access to tables and views.")
+    public String dataset;
+
+    @Name(NAME_BUCKET)
+    @Macro
+    @Nullable
+    @Description("The Google Cloud Storage bucket to store temporary data in. "
+            + "Cloud Storage data will be deleted after it is loaded into BigQuery. "
+            + "If it is not provided, a unique bucket will be automatically created and then deleted after the run " +
+            "finishes. The service account must have permission to create buckets in the configured project.")
+    public String bucket;
+
+    @Name(NAME_CMEK_KEY)
+    @Macro
+    @Nullable
+    @Description("The GCP customer managed encryption key (CMEK) name used to encrypt data written to " +
+            "any bucket created by the plugin. If the bucket already exists, this is ignored.")
```

Comment:
`bucket, dataset or table` created by the plugin in case of BQ plugins.

Reply:
I see, will update the description